### PR TITLE
Added PermissionsBoundary and FunctionRole optional parameters

### PR DESCRIPTION
**Pull request by haihongren**
- PermissionsBoundary allows additional security control when the new role is created for this Lambda
- FunctionRole parameter allows customer to provide a predefined role with required privileges

### CODE REVIEW

File: src/handler.py [hunk 1]
```diff
@@ -51,6 +50,41 @@ class BadRequestException(Exception):
...
+    return bool(re.search(regex_pattern, key))
+
+def _convert_float(s):
+    try:
+        f = float(s)
+    except ValueError:
+        f = 1.5
+    return f
+
+def _get_batch_size_factor(batch_size_factor=None):
```
Comment (William-Hill) [inline src/handler.py:80]: Why doesn't the `BATCH_SIZE_FACTOR` returned from this function fall back to the value we had before if it's not provided as an environment variable?

Comment (William-Hill) [inline src/handler.py:80]: Why doesn't the `BATCH_SIZE_FACTOR` returned from this function fall back to the value we had before if it's not provided as an environment variable?

File: src/handler.py [hunk 2]
```diff
@@ -11,15 +11,14 @@
 import logging
 from smart_open import open
 import re
-from pympler import asizeof
```
Comment (William-Hill) [inline src/handler.py:14]: @haihongren Why was the use of `asizeof` removed? `sys.getsizeof()` returns the reference size of the data structure in but doesn't recursively look at the size of the data structure's content. `asizeof` can give a more accurate measure.

Reference: https://stackoverflow.com/a/52523163

Comment (haihongren) [reply src/handler.py:14]: @William-Hill The `asizeof`  did not always properly.  My customer still ran into payload size exceeding limit error frequently with `asizeof`.  I was able to reproduce the behaviour with their log files. 

I explored various methods and found when I  converted  the log entry to string using `str(log)`, then `sys.getsizeof(str(log))`.  It worked consistently.   My customer has been using the updated code for past 8+ months and no problem reported so far.

**Review by William-Hill — DISMISSED** [review]
Thanks for the comments and updates @haihongren

File: template.yml [hunk 3]
```diff
@@ -1,6 +1,31 @@
 AWSTemplateFormatVersion: '2010-09-09'
 Transform: AWS::Serverless-2016-10-31
 Description: Sends log data from S3 to New Relic Logging.
+Metadata:
+  AWS::ServerlessRepo::Application:
+    Name: NewRelic-log-ingestion-s3-RPB
```
Comment (danybmx) [inline template.yml:6]: Is there any reason to change the application name?

File: template.yml [hunk 4]
```diff
@@ -1,6 +1,31 @@
...
+    Description: Send log data from a S3 bucket to New Relic Logging.
+    Author: New Relic
+    SpdxLicenseId: Apache-2.0
+    LicenseUrl: LICENSE
+    ReadmeUrl: README.md
+    Labels: ['newrelic', 'logs', 'logging', 'ingestion', 'lambda', 's3']
+    HomePageUrl: https://github.com/newrelic/aws_s3_log_ingestion_lambda
+    SemanticVersion: 1.1.10
```
Comment (danybmx) [inline template.yml:14]: From where this SemanticVersion comes? I think it should be 1.1.3 (I just bumped it on your PR a few minutes ago)

**Review by danybmx — CHANGES_REQUESTED** [review]
Hi @haihongren,

First of all thanks for your contribution and sorry for bothering. I need to bump the version to 1.1.3 a few minutes ago on this PR becase we merged a previous one (that only contains a version bump that was not properly done on latest change). Also added some comments on the template file to get a better understand on the changes.

Regards and thank you again for your contribution.

File: template.yml (hunk 3 above)
Comment (haihongren) [reply template.yml:6]: thanks. I was testing my own copy of the application - where I changed the application name. 
The application name should remain for the official release.

File: template.yml (hunk 4 above)
Comment (haihongren) [reply template.yml:14]: thanks. I was testing my own copy of the application - where I changed the application name. 
The versioning should follow the official cadence.

**Comment by danybmx** [comment]
Thanks for the clarification @haihongren, I've made the changes and will be merging and publishing this in a few.

**Review by danybmx — APPROVED** [review]

**Pull request merged**